**Importing the Libraries**

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

import nltk

from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import LogisticRegression
from sklearn.naive_bayes import MultinomialNB
from sklearn.metrics import (
    accuracy_score,
    precision_score,
    recall_score,
    f1_score,
    confusion_matrix,
    classification_report
)

**Download NLTK resources**

In [2]:
nltk.download('stopwords')
nltk.download('punkt')
nltk.download('wordnet')
nltk.download('omw-1.4')

[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\MTB\AppData\Roaming\nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\MTB\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package wordnet to
[nltk_data]     C:\Users\MTB\AppData\Roaming\nltk_data...
[nltk_data]   Package wordnet is already up-to-date!
[nltk_data] Downloading package omw-1.4 to
[nltk_data]     C:\Users\MTB\AppData\Roaming\nltk_data...
[nltk_data]   Package omw-1.4 is already up-to-date!


True

**Loading the dataset**

In [3]:
df = pd.read_csv("../dataset/amazon_alexa.tsv", sep="\t")
df.head()

,rating,date,variation,verified_reviews,feedback
0,5,31-Jul-18,Charcoal Fabric,Love my Echo!,1
1,5,31-Jul-18,Charcoal Fabric,Loved it!,1
2,4,31-Jul-18,Walnut Finish,"Sometimes while playing a game, you can answer...",1
3,5,31-Jul-18,Charcoal Fabric,I have had a lot of fun with this thing. My 4 ...,1
4,5,31-Jul-18,Charcoal Fabric,Music,1


### Data Exploration Section

**Displaying first 5 rows**

In [4]:
df.head()

,rating,date,variation,verified_reviews,feedback
0,5,31-Jul-18,Charcoal Fabric,Love my Echo!,1
1,5,31-Jul-18,Charcoal Fabric,Loved it!,1
2,4,31-Jul-18,Walnut Finish,"Sometimes while playing a game, you can answer...",1
3,5,31-Jul-18,Charcoal Fabric,I have had a lot of fun with this thing. My 4 ...,1
4,5,31-Jul-18,Charcoal Fabric,Music,1


**Displaying Last 5 rows**

In [5]:
df.tail()

,rating,date,variation,verified_reviews,feedback
3145,5,30-Jul-18,Black Dot,"Perfect for kids, adults and everyone in betwe...",1
3146,5,30-Jul-18,Black Dot,"Listening to music, searching locations, check...",1
3147,5,30-Jul-18,Black Dot,"I do love these things, i have them running my...",1
3148,5,30-Jul-18,White Dot,Only complaint I have is that the sound qualit...,1
3149,4,29-Jul-18,Black Dot,Good,1


**Dataset Shape**

In [6]:
print("Dataset Shape:", df.shape)

Dataset Shape: (3150, 5)


**Displaying Column Names**

In [7]:
print(df.columns)

Index(['rating', 'date', 'variation', 'verified_reviews', 'feedback'], dtype='object')


**Displaying Dataset Information**

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3150 entries, 0 to 3149
Data columns (total 5 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   rating            3150 non-null   int64 
 1   date              3150 non-null   object
 2   variation         3150 non-null   object
 3   verified_reviews  3149 non-null   object
 4   feedback          3150 non-null   int64 
dtypes: int64(2), object(3)
memory usage: 123.2+ KB


Displaying Statistical Summary of the Dataset

In [9]:
df.describe()

,rating,feedback
count,3150.000000,3150.000000
mean,4.463175,0.918413
std,1.068506,0.273778
min,1.000000,0.000000
25%,4.000000,1.000000
50%,5.000000,1.000000
75%,5.000000,1.000000
max,5.000000,1.000000


Displaying Statistical Summary Including Text Columns

In [10]:
df.describe(include='all')

,rating,date,variation,verified_reviews,feedback
count,3150.000000,3150,3150,3149,3150.000000
unique,NaN,77,16,2300,NaN
top,NaN,30-Jul-18,Black Dot,,NaN
freq,NaN,1603,516,79,NaN
mean,4.463175,NaN,NaN,NaN,0.918413
std,1.068506,NaN,NaN,NaN,0.273778
min,1.000000,NaN,NaN,NaN,0.000000
25%,4.000000,NaN,NaN,NaN,1.000000
50%,5.000000,NaN,NaN,NaN,1.000000
75%,5.000000,NaN,NaN,NaN,1.000000


Displaying Missing Values

In [11]:
print(df.isnull().sum())

rating              0
date                0
variation           0
verified_reviews    1
feedback            0
dtype: int64


Displaying Duplicate Records

In [12]:
print("Duplicate Rows:", df.duplicated().sum())

Duplicate Rows: 715


Displaying Unique Values

In [13]:
for column in df.columns:
    print(f"\n{column}")
    print(df[column].nunique())


rating
5

date
77

variation
16

verified_reviews
2300

feedback
2


Displaying Random Reviews

In [14]:
df.sample(5)

,rating,date,variation,verified_reviews,feedback
257,5,29-Jul-18,Charcoal Fabric,Very pleased with the Echo 2nd Generation. It ...,1
2680,5,30-Jul-18,White Dot,Easy set up,1
106,5,30-Jul-18,Charcoal Fabric,Very easy to set up. Works wonderful and I lov...,1
747,5,30-Jul-18,Charcoal Fabric,Works as you’d expect and then some. Also good...,1
1911,4,26-Jul-18,Black Plus,Still needs some work on commands. Sound is eq...,1


### Observations

- The dataset contains approximately 3,150 reviews.
- There are 5 columns.
- The target variable is `feedback`.
- Most columns contain no missing values.
- Duplicate reviews may be present and will be handled during preprocessing.
- The review text is stored in the `verified_reviews` column.